test/ngAnimate/animateRunnerSpec.js

```diff
@@ -0,0 +1,317 @@
+        var animationComplete = false;
+        runner.finally(function() {
+          animationComplete = true;
+        });
+        runner[method]();
+        $rootScope.$digest();
+        expect(animationComplete).toBe(true);
+    });
+  });
+
+  describe(".all()", function() {
+    it("should resolve when all runners have naturally resolved",
+      inject(function($$rAF, $$AnimateRunner) {
+
+      var runner1 = new $$AnimateRunner();
+      var runner2 = new $$AnimateRunner();
+      var runner3 = new $$AnimateRunner();
+
+      var status;
+      $$AnimateRunner.all([runner1, runner2, runner3], function(response) {
+        status = response;
+      });
+
+      runner1.complete(true);
+      runner2.complete(true);
+      runner3.complete(true);
+
+      expect(status).toBeUndefined();
+
+      $$rAF.flush();
+
+      expect(status).toBe(true);
+    }));
+
+    they("should immediately resolve if and when all runners have resolved have been $prop",
+      { ended: 'end', cancelled: 'cancel' }, function(method) {
+
+      inject(function($$rAF, $$AnimateRunner) {
+        var runner1 = new $$AnimateRunner();
+        var runner2 = new $$AnimateRunner();
+        var runner3 = new $$AnimateRunner();
+
+        var expectedStatus = method === 'end' ? true : false;
+
+        var status;
+        $$AnimateRunner.all([runner1, runner2, runner3], function(response) {
+          status = response;
+        });
+
+        runner1[method]();
+        runner2[method]();
+        runner3[method]();
+
+        expect(status).toBe(expectedStatus);
+      });
+    });
+
+    it("should return a status of `false` if one or more runners was cancelled",
+      inject(function($$rAF, $$AnimateRunner) {
+
+      var runner1 = new $$AnimateRunner();
+      var runner2 = new $$AnimateRunner();
+      var runner3 = new $$AnimateRunner();
+
+      var status;
+      $$AnimateRunner.all([runner1, runner2, runner3], function(response) {
+        status = response;
+      });
+
+      runner1.end();
+      runner2.end();
+      runner3.cancel();
+
+      expect(status).toBe(false);
+    }));
+  });
+
+  describe(".chain()", function() {
+    it("should evaluate an array of functions in a chain",
+      inject(function($$rAF, $$AnimateRunner) {
+
+      var runner1 = new $$AnimateRunner();
+      var runner2 = new $$AnimateRunner();
+      var runner3 = new $$AnimateRunner();
+
+      var log = [];
+
+      var items = [];
+      items.push(function(fn) { log.push(1); runner1.done(fn); });
+      items.push(function(fn) { log.push(2); runner2.done(fn); });
+      items.push(function(fn) { log.push(3); runner3.done(fn); });
+
+      var status;
+      $$AnimateRunner.chain(items, function(response) {
+        status = response;
+      });
+
+      runner1.complete(true);
+      runner2.complete(true);
+      runner3.complete(true);
+      $$rAF.flush();
+
+      expect(log).toEqual([1,2,3]);
+      expect(status).toBe(true);
+    }));
```

Comment:
OK updated. Refactoring this test brought up an issue where I referenced `this.state` instead of `this._state`. Thanks for reminding me again about this.